week-3/Homework/mandatory/4-eligible-students.js

```diff
@@ -7,8 +7,17 @@
   - Returns an array containing only the names of the who have attended AT LEAST 8 classes
 */
 
-function eligibleStudents() {
+function eligibleStudents(studentList) {
+  let newAttendedList = [];
+  // Loop through each student
+  for (let i = 0; i < studentList.length; i++) {
+    // Check second index is greater than 8 or more
+    if (studentList[i][1] >= 8) {
+      newAttendedList.push(studentList[i][0]);
+    }
+  }
 
+  return newAttendedList;
```

Comment:
Nice work! However, we could improve this a little by using an array method instead of a `for` loop. Which array method do you think would be useful here?

Reply:
`function eligibleStudents(studentList){
   return studentList.filter(studentAttendance => 
studentAttendance[1] >=8).map(studentName=>studentName[0])
  
 }`

Reply:
`function eligibleStudents(studentList){
return studentList.filter(studentAttendance =>
studentAttendance[1] >=8).map(studentName=>studentName[0])

}`

Reply:
function eligibleStudents(studentList){
return studentList.filter(studentAttendance =>
studentAttendance[1] >=8).map(studentName=>studentName[0])

`}```

Reply:
`function eligibleStudents(studentList){
return studentList.filter(studentAttendance =>
studentAttendance[1] >=8).map(studentName=>studentName[0])

}

Reply:
`function eligibleStudents(studentList){
return studentList.filter(studentAttendance =>
studentAttendance[1] >=8).map(studentName=>studentName[0])
}`

Reply:
Great!